addons/mail/models/mail_cc_mixin.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models, tools
+
+
+class MailCCMixin(models.AbstractModel):
+    _name = 'mail.cc.mixin'
+    _inherit = 'mail.thread'
+    _description = 'Email CC management'
+
+    email_cc = fields.Char('Email cc', help='List of cc from incoming emails.')
+
+    def _email_dict(self, cc_string):
+        '''return a dict of sanitize_email:raw_email from a string of cc'''
+        if not cc_string:
+            return {}
+        return {tools.email_split(cc)[0]: cc for cc in [cc.strip() for cc in cc_string.split(',')] if tools.email_split(cc)}
+
+    @api.model
+    def message_new(self, msg_dict, custom_values=None):
+        if custom_values is None:
+            custom_values = {}
+        cc_values = {
+            'email_cc': ", ".join(self._email_dict(msg_dict.get('cc')).values()),
+        }
+        cc_values.update(custom_values)
+        return super(MailCCMixin, self).message_new(msg_dict, cc_values)
+
+    @api.multi
+    def message_update(self, msg_dict, update_vals=None):
+        '''Adds cc email to self.email_cc while trying to keep email as raw as possible but unique'''
+        if update_vals is None:
+            update_vals = {}
+        cc_values = {}
+        new_cc = self._email_dict(msg_dict.get('cc'))
+        if new_cc:
+            old_cc = self._email_dict(self.email_cc)
```

Comment:
maybe I would put a comment that you make the union of `self.email_cc` and `msg_dict.get('cc')`

Reply:
it is explained in doc string